brownie/_cli/run.py

```diff
@@ -40,11 +40,12 @@ def main():
         active_project.load_config()
         print(f"{active_project._name} is the active project.")
 
-    network.connect(CONFIG.argv["network"])
-
     path, _ = _get_path(args["<filename>"])
     path_str = path.absolute().as_posix()
 
+    network_from_script = _get_network_from_script(path)
```

Comment:
One potential issue here - within `_get_network_from_script` you call to `_import_path`, which means the module gets imported prior to connecting to the network.  If anything happens in the global namespace that requires an active network connection, the script will now break.

The first solultion that comes to mind for me: instead of importing the module, parse the ast and then search for the related `Assign` node.  Something like this (tho for sure it'll need more validation):

```python
with path.open() as fp:
    module_ast = ast.parse(fp.read())
    network_name = next(
        (i.value.value for i in module_ast.body if isinstance(i, ast.Assign) and i.targets[0].id == "brownie_default_network"),
        None
    )
    if network_name is not None: ...
```

Reply:
Thank you for the review. I agree with your concern.

Parsing the ast could only help us search for the specific pattern of the assignment. Once it needs evaluation of the expression, it's hard to do in pure AST.